ballerina/websocket_service.bal

```diff
@@ -25,68 +24,73 @@ isolated service class WsService {
+                parser:FieldNode fieldNode = <parser:FieldNode>rootFieldVisitor.getRootFieldNode();
+                stream<any, error?>|json sourceStream = getSubscriptionResponse(self.engine, self.schema,
+                                                                                self.context, fieldNode);
+                if sourceStream is stream<any, error?> {
+                    record {|any value;|}|error? next = sourceStream.iterator().next();
+                    while next !is error? {
+                        ExecutorVisitor executor = new(self.engine, self.schema, self.context, {}, next.value);
+                        OutputObject outputObject = executor.getExecutorResult(node);
+                        ResponseFormatter responseFormatter = new(self.schema);
+                        OutputObject coercedOutputObject = responseFormatter.getCoercedOutputObject(outputObject, node);
+                        if coercedOutputObject.hasKey(DATA_FIELD) || coercedOutputObject.hasKey(ERRORS_FIELD) {
+                            check caller->writeTextMessage(coercedOutputObject.toString());
+                        }
+                        next = sourceStream.iterator().next();
+                    }
+                } else {
+                    check caller->writeTextMessage(sourceStream.toJsonString());
                 }
+            } else {
+                check caller->writeTextMessage((<ErrorDetail>node).message);
             }
-            check caller->writeTextMessage((<ErrorDetail>node).message);
         }
     }
 }
 
 isolated function validateSubscriptionPayload(string text, Engine engine) returns parser:OperationNode|ErrorDetail {
-    json payload = text.toJson();
-    var document = payload.query;
-    var variables = payload.variables;
-    variables = variables is error ? () : variables;
-    if document is string && document != "" {
-        if variables is map<json> || variables is () {
-            parser:OperationNode|OutputObject validationResult = engine.validate(document, getOperationName(payload),
-                                                                                 variables);
-            if validationResult is parser:OperationNode {
-                return validationResult;
-            }
-            return {message: validationResult.toJsonString()};
-        }
-        return {message: "Invalid format in request parameter: variables"};
+    json|error payload = value:fromJsonString(text);
+    if payload is error {
+        json errorMessage = {errors: [{message: "Invalid subscription payload"}]};
+        return {message: errorMessage.toJsonString()};
+    }
+    json|error document = payload.query;
+    if document !is string || document == "" {
+        json errorMessage = {errors: [{message: "Query not found"}]};
+        return {message: errorMessage.toJsonString()};
     }
-    return {message: "Query not found"};
+    json|error variables = payload.variables;
```

Comment:
Since the `variables field` is optional in standard GraphQL requests, this can be an `error` even for the valid requests.